### PR TITLE
feat(token-cache): optional logger

### LINKED ISSUE #152: TokenCache should support logger
Just like the other public functions of the lib (`getAccessToken`, `GetTokenInfo`, middlewares) it should be possible to pass a logger to `TokenCache`. Appropriate calls to the given logger should be added as well.

### PR DESCRIPTION
**Pull request by bzums**
close #152

### CODE REVIEW

File: src/token-cache.ts [hunk 1]
```diff
@@ -147,27 +166,34 @@ class TokenCache {
...
+      const msg = `TokenCache miss: ${tokenName}`;
+      this.logger.warn(msg);
+      return Promise.reject(msg);
     }
 
     const localExpiry = token.local_expiry || 0;
 
     if (localExpiry < Date.now()) {
-      return Promise.reject(`Token ${tokenName} expired locally.`);
+      const msg = `TokenCache miss: ${tokenName} expired locally`;
```
Comment (ISO50) [inline src/token-cache.ts]: Is there a reason to put the string first into msg?

File: src/token-cache.ts [hunk 2]
```diff
@@ -147,27 +166,34 @@ class TokenCache {
...
+      this.logger.warn(msg);
```
Comment (ISO50) [inline src/token-cache.ts]: really something we have to warn?

Comment (bzums) [reply src/token-cache.ts]: Mh yeah you are right, maybe rather `debug` right?

Comment (ISO50) [reply src/token-cache.ts]: 👍

**Comment by ISO50** [comment]
👍

**Comment by bzums** [comment]
👍

**Comment by Retro64** [comment]
:+1:

**Pull request merged**